app/workers/new_page_worker.rb

```diff
@@ -0,0 +1,42 @@
+class NewPageWorker
+  include Sidekiq::Worker
+
+  def perform(page_id)
+    page = Page.find(page_id)
+    page.update_attributes(status: Page::ACTIVE_STATUS)
+    parse_html_content(page)
+    make_screenshot(page)
+    page.update_attributes(status: Page::FINISHED_STATUS)
+    logger.info "Page #{page.id} processed!"
+  rescue SocketError, Gastly::FetchError, OpenURI::HTTPError, Net::OpenTimeout => e
+    page.update_attributes(status: Page::ERROR_STATUS)
+    logger.warn e
+    return
```

Comment:
no need

Reply:
How can I handle errors if page_url aren't valid? Worker will crash and retry to parse content.

Reply:
yes, handling errors is needed